tests/e2e/support/objects/app-files/share/actions.ts

```diff
@@ -95,6 +99,26 @@ export const acceptShare = async (args: ShareStatusArgs): Promise<void> => {
   await page.locator(util.format(filesSharedWithMeAccepted, resource)).waitFor()
 }
 
+export const acceptAllShare = async ({ page }: { page: Page }): Promise<void> => {
+  if (await page.isVisible(showAllButton)) {
+    await page.locator(showAllButton).click()
+  }
+  await page.locator(selecAllCheckbox).click()
+  const numberOfPendingShares = await page.locator(numberOfShares).count()
+  const checkResponses = []
+  for (let i = 0; i < numberOfPendingShares; i++) {
+    checkResponses.push(
+      page.waitForResponse(
+        (resp) =>
+          resp.url().includes('shares') &&
+          resp.status() === 200 &&
+          resp.request().method() === 'POST'
+      )
+    )
+  }
```

Comment:
You can try this one:
```suggestion
const declinedShareItems = '//div[@id="files-shared-with-me-pending-section"]//tr[contains(@class,"oc-tbody-tr")]'
...
  for (let i = 0; i < numberOfPendingShares; i++) {
    const id = await page.locator(declinedShareItems + `[${i + 1}]`).getAttribute('data-item-id')
    checkResponses.push(
      page.waitForResponse(
        (resp) =>
          resp.url().includes(`shares/pending/${id}`) &&
          resp.status() === 200 &&
          resp.request().method() === 'POST'
      )
    )
  }
```

Reply:
But it could be flaky if `page.locator(numberOfShares).count()` doesn't give exact number consistently

Reply:
I see that sometimes the count can be less than the actual but test will pass because the provided number of shares were accepted but in such case we will not know that all shares were accepted successfully

Reply:
I found my error https://github.com/owncloud/web/pull/9734/files#diff-bdf82996a7fca0461152dfa6620089e336592e6a02b7cb5854e1d3fab3c1cd3bR104
count was less because test doesn't click button "show more"

Regarding checking id. That is fine, thanks for suggestion